### DISCUSSION TITLE
Utilisation for the working time of employees

### DISCUSSION THREAD

**Discussion by Alex-Zwenkau** [te-online/timemanager#167] [General]
In Germany, we have the problem that we will need electronic working time recording for employees and workers in the future. In my opinion, this could be done well with TimeManger and the mobile app.

However, this would require some changes: One should perhaps not speak of *customers, projects* or *tasks* to employees and workers. There should be an option to rename these 3 areas. 

And perhaps also to avoid errors caused by this, this renaming should be separately adjustable in the administration for each individual server instance. 

It should also be possible to centrally pre-assign the variables *customers, project* and *tasks*, as this usually remains unchanged for employees and workers over very long periods of time. Changes are usually specified by the company and could therefore be made via the administration.

**Comment by te-online** [comment]
Hey there 👋  Thanks for your question!

- You can't rename the areas right now and I have no plans to add a feature to rename them. I can see how that would be nice, but it's not something that is on my roadmap. You're welcome to add it yourself, of course 😊 

The overwhelming majority of businesses that track time work for clients, on projects where each project can be split into tasks. If your business is different that's a problem of course, sorry I don't have a better answer.

> It should also be possible to centrally pre-assign the variables customers, project and tasks, as this usually remains unchanged for employees

- It sounds to me like you're looking for a different kind of software, because you really want an attendance clock. 
- Keep in mind that only an [app for Android exists and that one is no longer maintained](https://gitlab.com/joleaf/nc-timemanager-mobile-app). However, you should be able to use the app as a PWA within a browser within Nextcloud, I assume.
- If you go with this app, which I can't really recommend as there seems to be a mismatch between your requirements and the available features, I can recommend to have a manager create a client, with a pre-defined project and task. Share this client with your employees and have them track the time in the pre-defined entities. If they create their own clients, they can do that, but you can only count whatever is in the shared client/project/task.

I'll convert this to a discussion, as I don't really see any actionable items for me here 😊

**Reply by Alex-Zwenkau** [reply]
Thank you very much for your answers. You have correctly recognised that I am looking for an app for a kind of time clock with which I can easily save the results per employee and month directly in the Nextcloud.
I can understand that you see it differently and that TimeManager is probably not suitable.
Unfortunately, there is currently no app that allows you to track time for your own employees in Nextcloud. I am very surprised by this, as Nextcloud is widely used within companies, authorities and administrations.
So I'll have to keep looking or find someone who can programme it specifically.

**Comment by dustbro** [comment]
Just starting to utilize TimeManager. I would love to see this app expanded to include Group and User assignments. In a perfect world, Admins would be able to assign a task to a Group or User so those people can track their time on company-wide projects and tasks.
